core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -66,6 +72,47 @@
 
   private static final Logger log = LoggerFactory.getLogger(ApplicationServer.class);
 
+  private static final Version VERSION =
+      parseVersion(System.getProperty("java.specification.version"));
+
+  // Package private for testing
+  static Version parseVersion(String versionString) {
+    final StringTokenizer st = new StringTokenizer(versionString, ".");
+    int majorVersion = Integer.parseInt(st.nextToken());
+    int minorVersion;
+    if (st.hasMoreTokens()) {
+      minorVersion = Integer.parseInt(st.nextToken());
+    } else {
+      minorVersion = 0;
+    }
+    return new Version(majorVersion, minorVersion);
+  }
+
+  // Having these as static final provides the best opportunity for compiler optimization
+  public static final boolean IS_JAVA11_COMPATIBLE = VERSION.isJava11Compatible();
+
+  // Package private for testing
```

Comment:
nit: Are there test changes that were supposed to be part of the PR? I don't think I'm seeing any. Anyway, see my next comment below.

Reply:
Yes, I agree. Having discuss with @rigelbm, I'll add a test.

Reply:
+1, waiting on the test that confirms HTTP/2 works, since I believe the HTTP client we currently use in our tests does not support HTTP/2 at all.

Reply:
Tests added.